### DISCUSSION TITLE
Program never progresses

### DISCUSSION THREAD

**Discussion by EclecticInventor** [pybricks/support#1218] [Q&A] [answered]
Using the latest beta, v3.3.0b8 (Pybricks Beta v2.2.0-beta.8), the code never detects white. running the example color detection code shows the sensor is detecting white. Is this a bug or something I'm missing?
```python
from pybricks.hubs import InventorHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor, ColorDistanceSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.tools import wait, StopWatch, run_task, multitask

gameStatus = 0

hub = InventorHub()
hub.display.orientation(up=Side.RIGHT)
coinSensor = ColorSensor(Port.A)

async def coinScan():
    while coinSensor.color() != Color.WHITE:
        await wait(20)
    print ("white")

async def attractMode():
    await multitask(coinScan())

#attract mode
run_task(attractMode())
gameStatus = 1
```

**Comment by laurensvalk** [comment, marked as answer]
All sensor methods are awaitables, too.

Change this:

```python
coinSensor.color()
```

to

```python
await coinSensor.color()
```

An upcoming documentation update will include `await` on each method where it is needed to make this easier to see.

Thanks!

**Reply by EclecticInventor** [reply]
I tried changing it to 
await coinSensor.color() == Color.WHITE

and removing the wait but it just skips over it.

**Reply by laurensvalk** [reply]
Why remove the wait? That would leave an empty loop.

Happy to have a look at your new script :)

**Reply by laurensvalk** [reply]
I also noticed you changed != to ==, so it would only wait while white. It it doesn’t see white, the loop ends right away.

If in doubt, try a test program without await and multitasking first.

**Reply by EclecticInventor** [reply]
I just figured it out. I had it as await while, instead of while await
